manager-bundle/src/EventListener/DoctrineAlterTableListener.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\ManagerBundle\EventListener;
+
+use Doctrine\DBAL\Event\SchemaAlterTableRenameColumnEventArgs;
+use Doctrine\DBAL\Schema\TableDiff;
+
+/**
+ * @internal
+ */
+class DoctrineAlterTableListener
+{
+    /**
+     * Prevents renaming arbitrary columns by explicitly dropping the old ones
+     * and adding the new ones (see #1716).
+     */
+    public function onSchemaAlterTableRenameColumn(SchemaAlterTableRenameColumnEventArgs $args)
+    {
+        $args->preventDefault();
+
+        $platform = $args->getPlatform();
+        $table = $args->getTableDiff()->getName($platform);
+        $oldColumn = $args->getTableDiff()->fromTable->getColumn($args->getOldColumnName());
+        $column = $args->getColumn();
+
+        $tableDiff = new TableDiff($table->getName());
+        $tableDiff->removedColumns[$args->getOldColumnName()] = $oldColumn;
```

Comment:
You could probably use the new `$column` here as well as Doctrine only uses the key, but not sure. Not important either :D